### PR TITLE
ID-363 let Sam host privacy policy in tandem with ToS

### PR DESCRIPTION
**Pull request by tlangs**
This PR can't merge until https://github.com/broadinstitute/sam/pull/920 does.

Currently, Sam hosts the Terms of Service, but the UI hosts the privacy policy. Now that we are faced with updating both the Terms of Service and Privacy Policy at the same time, it makes sense that both would be stored in the same system. This change will allow us to have different Privacy Policies on prod and pre-prod envs, since Sam will now serve the privacy policy version based on its config. It also means the Terms of Service and Privacy Policy are always in sync. No need to coordinate multiple PRs to update future documents!

### CODE REVIEW

File: config/dev.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "orchestrationUrlRoot": "https://firecloud-orchestration.dsde-dev.broadinstitute.org",
   "rawlsUrlRoot": "https://rawls.dsde-dev.broadinstitute.org",
   "rexUrlRoot": "https://terra-rex-dev.appspot.com",
-  "samUrlRoot": "https://sam.dsde-dev.broadinstitute.org",
+  "samUrlRoot": "https://local.broadinstitute.org:50443",
```
Comment (nawatts) [inline config/dev.json]: Development config that shouldn't be merged.

File: src/pages/PrivacyPolicy.js [hunk 2]
```diff
@@ -1,114 +1,37 @@
...
-      style: { maxWidth: '45rem' },
-      renderers: {
-        link: newWindowLinkRenderer
-      }
-    }, [privacyPolicy])
+  const [privacyPolicy, setPrivacyPolicy] = useState()
+
+  useOnMount(() => {
+    const loadPrivacyPolicyAndUpdateState = (async () => {
+      setPrivacyPolicy(await Ajax().User.getPrivacyPolicy())
```
Comment (nawatts) [inline src/pages/PrivacyPolicy.js]: Probably worth some error handling on this API request, at least wrapping with `withErrorReporting` to show a notification if the request fails.

File: config/dev.json (hunk 1 above)
Comment (cahrens) [inline config/dev.json]: You don't want to check this in.

File: src/pages/PrivacyPolicy.js [hunk 3]
```diff
@@ -1,114 +1,37 @@
...
+    })
+    loadPrivacyPolicyAndUpdateState()
+  })
+
+  return div({ role: 'main', style: { padding: '1rem', minHeight: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' } }, [
```
Comment (nawatts) [inline src/pages/PrivacyPolicy.js]: Could the style changes be a separate PR? Also, it would be nice to have before/after screenshots with visual changes like this.

File: src/libs/ajax.js [hunk 4]
```diff
@@ -127,6 +127,11 @@ const User = signal => ({
     }
   },
 
+  getPrivacyPolicy: async () => {
+    const response = await fetchSam('privacy/text', _.merge(authOpts(), { signal }))
```
Comment (nawatts) [inline src/libs/ajax.js:131]: Is `authOpts` required here? Shouldn't users be able to view the privacy policy before logging in?

File: config/dev.json (hunk 1 above)
Comment (tlangs) [reply config/dev.json]: Totally right

File: src/libs/ajax.js (hunk 4 above)
Comment (tlangs) [reply src/libs/ajax.js:131]: Good point. They should.

Comment (tlangs) [reply src/libs/ajax.js:131]: tested locally, and can still access the privacy policy even if not authed. Sam doesn't require auth for this new endpoint.

File: src/pages/PrivacyPolicy.js (hunk 3 above)
Comment (tlangs) [reply src/pages/PrivacyPolicy.js]: Removed styling changes

File: src/pages/PrivacyPolicy.js (hunk 2 above)
Comment (tlangs) [reply src/pages/PrivacyPolicy.js]: Added error handling.

File: config/dev.json (hunk 1 above)
Comment (tlangs) [reply config/dev.json]: rectified

Comment (tlangs) [reply config/dev.json]: rectified

File: src/libs/ajax.js (hunk 4 above)
Comment (nawatts) [reply src/libs/ajax.js:131]: If this endpoint doesn't require auth, then I'd suggest removing `authOpts` from this request.

**Review by nawatts — APPROVED** [review]

**Comment by tlangs** [comment]
@nawatts by including authopts, we can see _who_ is looking at the Privacy Policy, if they're logged in. Seems like it might be useful, but can remove if its gonna cause problems.

**Comment by tlangs** [comment]
The tests are failing for a completely separate reason from this PR.

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by tlangs** [comment]
Merging because tests are failing for an unrelated reason
